Luciefifi/my-brand-backend-api
# Ft blogtesting 013

# What does this PR do 
Unit testing of all blog endpoints using the Mocha and Chai testing framework. 

#Description of the task to be complete
- Add test cases for all the blog endpoints using Mocha and Chai
The tests will cover the following scenarios:
- Retrieving a list of all blog posts
- Retrieving a specific blog post by ID
- Creating a new blog post
- Updating an existing blog post
- Deleting a blog post
These tests will ensure that the endpoints are working correctly and that any future changes to the codebase will not break existing functionality.


# How should this be manually tested 
after cloning the repository cd into it and the following 
- npm install to install all node_modules , after that go ahead and start testing if the test are working by running npm test in your terminal 
- when the test passed or failed you will clearly see it . you will see the number of failed and passed test


# any background context I want to provide  
Mocha is not the only nodeJs unit testing framework 
There are many other framework that can be used in testing . those includes : jest , jasmine , AVA.

#related pivotal tracker user story
#13
